[build] Better detect initial build by checking for presence of CMakeCache.txt, fixes #53

bwasti opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#56 [build] Better detect initial build by checking for presence of CMakeCache.txt, fixes #53**

bwasti committed:
[build] Better detect initial build by checking for presence of CMakeCache.txt, fixes #53

bddppq commented on setup.py:
```diff
@@ -166,8 +166,8 @@ def _run_build(self):
     def run(self):
-        is_initial_build = not os.path.exists(CMAKE_BUILD_DIR)
-        if is_initial_build:
+        is_initial_build = not os.path.exists(CMAKE_BUILD_DIR) or not os.path.exists(os.path.join(CMAKE_BUILD_DIR, "CMakeCache.txt"))
```
Suggestion:
```python
if not os.path.exists(CMAKE_BUILD_DIR):
    os.makedirs(CMAKE_BUILD_DIR)

is_initial_build = not os.path.exists(os.path.join(CMAKE_BUILD_DIR, "CMakeCache.txt"))
if is_initial_build or RERUN_CMAKE:
    self._run_cmake()
```

bwasti committed:
Update on "[build] Better detect initial build by checking for presence of CMakeCache.txt, fixes #53"

[build] Better detect initial build by checking for presence of CMakeCache.txt, fixes #53

gh-metadata: pytorch tvm 56 gh/bwasti/37/head
